src/middlewared/middlewared/plugins/system.py

```diff
@@ -152,6 +156,335 @@ def debug(self, job):
         return dump
 
 
+class GeneralSystemService(ConfigService):
+
+    class Config:
+        datastore = 'system.settings'
+        datastore_prefix = 'stg_'
+
+    def __init__(self, *args, **kwargs):
+        super(GeneralSystemService, self).__init__(*args, **kwargs)
+        self._languages = self._initialize_system_languages()
+        self._time_zones_list = None
+        self._kbdmap_choices = None
+
+    @accepts()
+    def get_system_languages(self):
+        return self._languages
+
+    def _initialize_system_languages(self):
+        LANGUAGES = [
```

Comment:
Perhaps it makes sense to make it a dict instead? Easier to lookup, no need to iterate over all.

Reply:
Also, no need to make it uppercase (LANGUAGES)